nvflare/app_common/resource_managers/list_resource_manager.py

```diff
@@ -12,66 +12,138 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
+import time
 import uuid
+from collections import deque
+from threading import Event, Lock, Thread
 from typing import Dict, List, Optional
 
+from nvflare.apis.event_type import EventType
+from nvflare.apis.fl_component import FLComponent
 from nvflare.apis.fl_context import FLContext
 from nvflare.apis.resource_manager_spec import ResourceManagerSpec
 
 
-class ListResourceManager(ResourceManagerSpec):
+class ListResourceManager(ResourceManagerSpec, FLComponent):
     """Manage a list of resource units.
 
     For example:
 
-        - require 2, resource is [0, 1, 2, 3, 4, 5] -> check if things in list is available => return [0,1]
-        - require 3 => return [2, 3, 4]
-        - free 2, require 3 => return [0, 1, 5]
+        - require 2, current resources is [0, 1, 2, 3, 4, 5] => return [0,1]
+          after allocation the current resources become [2, 3, 4, 5]
+        - require 3, current resources [2, 3, 4, 5] => return [2, 3, 4]
 
     """
 
-    def __init__(self, resources: Dict[str, List]):
+    def __init__(self, resources: Dict[str, List], expiration_period: int = 600):
```

Comment:
is there a central configuration to define all default values ? or we just hard-code the default value directly ?

Reply:
From what I know, we don't have a "central" config.